### DISCUSSION TITLE
The authorization grant type is not supported by the authorization server.

### DISCUSSION THREAD

**Discussion by slk1107** [timdorr/tesla-api#547] [Q&A] [answered]
The Api is broken:
URL: https://owner-api.teslamotors.com/oauth/token
Method: POST
Header: Bearer {access token from Step 3}
BODY:
"grant_type": "urn:ietf:params:oauth:grant-type:jwt-bearer"
"client_id": "81527cff06843c8634fdc09e8ac0abefb46ac849f38fe1e431c2ef2106796384"
"client_secret": "c7257eb71a564034f9419ee651c7d0e5f7aa6bfbd18bafb5c5c033b093bb2fa3"

Response:
{
    "error": "unsupported_grant_type",
    "error_description": "The authorization grant type is not supported by the authorization server."
}

Has anyone encounter the same problem?

**Comment by davidhodge** [comment, marked as answer]
Word was they were imminently going to disable v2 tokens. That may be what happened here.

**Reply by slk1107** [reply]
Oh mine~
Does that mean the token will only be valid for 8 hours in the future?

**Reply by denouche** [reply]
Hello, this error also occurs when using a v3 token.
The error message suggest that the grant_type "urn:ietf:params:oauth:grant-type:jwt-bearer" is not supported anymore.
How do we get the new grant_type to use?

**Reply by slk1107** [reply]
> Hello, this error also occurs when using a v3 token. The error message suggest that the grant_type "urn:ietf:params:oauth:grant-type:jwt-bearer" is not supported anymore. How do we get the new grant_type to use?

According to your request params, you're talking about v2 tokens instead of v3. 
v3 tokens are gotten from the Step 3 in the [document](https://tesla-api.timdorr.com/api-basics/authentication).

**Comment by dacastro4** [comment]
Check this [issue](https://github.com/timdorr/tesla-api/issues/548).

**Reply by slk1107** [reply]
Yes the issue is posted after my discussion. But thank you for linking in!
